### ISSUE TITLE
PHP Fatal error:  Uncaught InvalidArgumentException: Could not get class storage

### ISSUE THREAD

**Issue by paragonie-scott** [vimeo/psalm#460]
I'm attempting to use Psalm on [Discretion](https://github.com/paragonie/discretion), and I'm getting an interesting fatal error:
```
scott@paragonie-test:~/dev/discretion$ vendor/bin/psalm
PHP Fatal error:  Uncaught InvalidArgumentException: Could not get class storage for ParagonIE\Discretion\Struct&ParagonIE\Discretion\Policies\Unique in /home/scott/dev/discretion/vendor/vimeo/psalm/src/Psalm/Provider/ClassLikeStorageProvider.php:21
Stack trace:
#0 /home/scott/dev/discretion/vendor/vimeo/psalm/src/Psalm/Checker/MethodChecker.php(360): Psalm\Provider\ClassLikeStorageProvider->get('ParagonIE\\Discr...')
#1 /home/scott/dev/discretion/vendor/vimeo/psalm/src/Psalm/Checker/Statements/Expression/Assignment/PropertyAssignmentChecker.php(252): Psalm\Checker\MethodChecker::methodExists(Object(Psalm\Checker\ProjectChecker), 'ParagonIE\\Discr...')
#2 /home/scott/dev/discretion/vendor/vimeo/psalm/src/Psalm/Checker/Statements/Expression/AssignmentChecker.php(342): Psalm\Checker\Statements\Expression\Assignment\PropertyAssignmentChecker::analyzeInstance(Object(Psalm\Checker\StatementsChecker), Object(PhpParser\Node\Expr\PropertyFetch), 'created', Object(PhpParser\Node\Expr\New_), Object(Psalm\Type\Union), Object(Psalm\Context))
#3 /home/scott/dev/di in /home/scott/dev/discretion/vendor/vimeo/psalm/src/Psalm/Provider/ClassLikeStorageProvider.php on line 21

Fatal error: Uncaught InvalidArgumentException: Could not get class storage for ParagonIE\Discretion\Struct&ParagonIE\Discretion\Policies\Unique in /home/scott/dev/discretion/vendor/vimeo/psalm/src/Psalm/Provider/ClassLikeStorageProvider.php:21
Stack trace:
#0 /home/scott/dev/discretion/vendor/vimeo/psalm/src/Psalm/Checker/MethodChecker.php(360): Psalm\Provider\ClassLikeStorageProvider->get('ParagonIE\\Discr...')
#1 /home/scott/dev/discretion/vendor/vimeo/psalm/src/Psalm/Checker/Statements/Expression/Assignment/PropertyAssignmentChecker.php(252): Psalm\Checker\MethodChecker::methodExists(Object(Psalm\Checker\ProjectChecker), 'ParagonIE\\Discr...')
#2 /home/scott/dev/discretion/vendor/vimeo/psalm/src/Psalm/Checker/Statements/Expression/AssignmentChecker.php(342): Psalm\Checker\Statements\Expression\Assignment\PropertyAssignmentChecker::analyzeInstance(Object(Psalm\Checker\StatementsChecker), Object(PhpParser\Node\Expr\PropertyFetch), 'created', Object(PhpParser\Node\Expr\New_), Object(Psalm\Type\Union), Object(Psalm\Context))
#3 /home/scott/dev/di in /home/scott/dev/discretion/vendor/vimeo/psalm/src/Psalm/Provider/ClassLikeStorageProvider.php on line 21
```
For reference:

```
scott@paragonie-test:~/dev/discretion$ vendor/bin/psalm --version
Psalm 0.3.82@a253ca68bcbb910eb70ca7554fa7e0783593b195
```

**Comment by muglug** [comment]
I need to add intersection type support to the docblock type parser. Shouldn’t take more than a couple of hours.

**Comment by muglug** [comment]
Sorry, nothing to do with docblocks. Reproducible with

```php
abstract class A {
  /** @var string|null */
  public $foo;
  
  public static function getFoo(): void {
    $a = new static();
    if ($a instanceof I) {}
    $a->foo = 5;
  }
}

interface I {}
```

**Comment by muglug** [comment]
Fixed in 5200ffa214d6815a68cde94f2150d67bbeec99c6
